roofit/roofitcore/src/RooProdPdf.cxx

```diff
@@ -729,14 +727,13 @@ void RooProdPdf::factorizeProduct(const RooArgSet& normSet, const RooArgSet& int
 
     // Make list of cross dependents (term is self contained for these dependents,
     // but components import dependents from other components)
-    RooArgSet* crossDeps = (RooArgSet*) intNoNormDeps->selectCommon(*normDeps);
+    RooArgSet* crossDeps = (RooArgSet*) depIntNoNormList[i].selectCommon(*normDeps);
```

Comment:
Many of those casts in RF don't make sense. I didn't look it up, but this one looks like one you don't need. And `unique_ptr` if you can, see above comment.

Reply:
Done, the cast was indeed not needed